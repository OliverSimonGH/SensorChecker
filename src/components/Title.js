import React, {Component} from 'react';

class Title extends Component {
    render() {
        return (
            <div className="page-header center">
                <h2>{this.props.title}</h2>
            </div>

        );
    }
}

export default Title;