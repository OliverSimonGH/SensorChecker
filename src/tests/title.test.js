import React from 'react';
import ReactDOM from "react-dom";
import Title from '../components/Title';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('Title Component', () => {
    it('check if component rendered', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Title />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('checks if title is rendered', () => {
        const wrapper = shallow(<Title title="Sensor Application"/>);
        const title = wrapper.find('h2').text();
        expect(title).toEqual('Sensor Application');
    })

    it('matches the snapshot', () => {
        const tree = renderer.create(<Title title="Sensor Application"/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

