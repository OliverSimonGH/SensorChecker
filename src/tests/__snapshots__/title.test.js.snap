// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component matches the snapshot 1`] = `
<div
  className="page-header center"
>
  <h2>
    Sensor Application
  </h2>
</div>
`;
